entity-framework/core/querying/user-defined-function-mapping.md

```diff
@@ -0,0 +1,154 @@
+
+The function definition is as follows:
+
+[!code-csharp[Main](../../../samples/core/Querying/UserDefinedFunctionMapping/Model.cs#HasTranslationFunctionConfiguration)]
+
+Note that [SqlExpressionFactory](/dotnet/api/microsoft.entityframeworkcore.query.sqlexpressionfactory) is used to construct a `SqlExpression` tree.
+
+Once we define the function, it can be used in the query. Instead of calling database function, EF Core will translate the method body directly into SQL based on the SQL expression tree constructed from the HasTranslation. The following LINQ query:
+
+[!code-csharp[Main](../../../samples/core/Querying/UserDefinedFunctionMapping/Program.cs#HasTranslationQuery)]
+
+Produces the following SQL:
+
+```sql
+SELECT [p].[PostId], [p].[BlogId], [p].[Content], [p].[Rating], [p].[Title]
+FROM [Posts] AS [p]
+WHERE [p].[PostId] < CASE
+    WHEN [p].[BlogId] > 3 THEN [p].[BlogId] - 3
+    ELSE 3 - [p].[BlogId]
+END
+```
+
+## Mapping a Queryable function to a table-valued gunction
+
+EF Core also supports mapping to a table-valued function using a user-defined CLR function returning an `IQueryable` of entity types, allowing EF Core to map TVFs with parameters. The process is similar to mapping a scalar user-defined function to a SQL function: we need a TVF in the database, a CLR function that is used in the LINQ queries, and a mapping between the two.
+
+As an example, we'll use a table-valued function that returns all posts having at least one comment that meets a given "Like" threshold:
+
+```sql
+CREATE FUNCTION dbo.PostsWithPopularComments(@likeThreshold int)
+RETURNS @posts TABLE
+(
+    PostId int not null,
+    BlogId int not null,
+    Content nvarchar(max),
+    Rating int not null,
+    Title nvarchar(max)
+)
+AS
+BEGIN
+    INSERT INTO @posts
+    SELECT[p].[PostId], [p].[BlogId], [p].[Content], [p].[Rating], [p].[Title]
+    FROM[Posts] AS[p]
+    WHERE(
+        SELECT COUNT(*)
+        FROM[Comments] AS[c]
+        WHERE([p].[PostId] = [c].[PostId]) AND([c].[Likes] >= @likeThreshold)) > 0
+
+    RETURN
+END
+```
+
+The CLR function signature is as follows:
+
+[!code-csharp[Main](../../../samples/core/Querying/UserDefinedFunctionMapping/Model.cs#QueryableFunctionDefinition)]
+
+> [!TIP]
+> The `FromExpression` call in the CLR function body allows for the function to be the root of the EF Core query, meaning it can be used instead of a regular DbSet.
+
+And below is the mapping:
+
+[!code-csharp[Main](../../../samples/core/Querying/UserDefinedFunctionMapping/Model.cs#QueryableFunctionConfigurationHasDbFunction)]
+
+> [!CAUTION]
+> Mapping to an `IQueryable` of entity types overrides the default mapping to a table for the DbSet. If necessary - for example when the entity is not keyless - mapping to the table must be specified explicitly using `ToTable` method.
+
+When the function is mapped, the following query:
```

Comment:
Add this as note in this section => Mention that TVF cannot use HasTranslation API